torch/csrc/autograd/python_function.cpp

```diff
@@ -685,48 +680,16 @@ int THPFunction_set_materialize_grads(THPFunction *self, PyObject *value, void *
   END_HANDLE_TH_ERRORS_RET(-1)
 }
 
-static PyObject *unpack_saved_variables(
-    THPFunction *self,
-    const std::function<PyObject*(const Variable&)>& unpack_fn)
+static PyObject *unpack_saved_variables(THPFunction *self)
 {
-  THPUtils_assert(!self->has_freed_buffers, ERR_BACKWARD_TWICE);
-  auto& saved_variables = self->saved_variables;
-  if (saved_variables.empty())
-    return PyTuple_New(0);
-
-  int num_saved = saved_variables.size();
-  THPObjectPtr saved(PyTuple_New(num_saved));
-  if (!saved)
-    return nullptr;
-  auto saved_for = self->cdata.lock();
-  // This is really a true assert, because we've already tested for the
-  // self->has_freed_buffers case at the beginning of this function:
-  // buffers are freed when PyNode dies; if the buffers are not freed,
-  // PyNode must be live.  (Note that the buffers could be freed
-  // even though the PyNode is live, but that doesn't matter here
-  // because we will never hit this line of code if the buffers are freed--
-  // and in any case saved_for will be non-NULL.)
-  TORCH_INTERNAL_ASSERT(saved_for);
-  for (int i = 0; i < num_saved; i++) {
-    auto unpacked_var = saved_variables[i].unpack(saved_for);
-    THPObjectPtr value;
-    if (!unpacked_var.defined()) {
-      Py_INCREF(Py_None);
-      value = Py_None;
-    } else {
-      value = unpack_fn(unpacked_var);
-    }
-    PyTuple_SET_ITEM(saved.get(), i, value.release());
-  }
-  return saved.release();
+  Py_INCREF(self->to_save);
```

Comment:
Maybe we'd need to replicate some of the saved variable's unpack logic here, i.e., detecting inplace modification and backward twice.

Reply:
Removing this change